docs/architecture/adr.rst

```diff
@@ -439,6 +439,190 @@ Conclusion
+* *crs* — ``{EPSG code | EPSG name | proj4 string}`` destination CRS
+
+*Tile definition:*
+
+.. code:: javascript
+
+ {
+    "type": "tile",
+    "maxZoom": 19,
+    "tileSize": 256,
+    "resampleMethod": "{nearest-neighbor | bilinear | cubic-convolution | cubic-spline | lanczos}",
+    "layoutScheme": "zoomed",
+    "keyIndexMethod": {
+      "type": "{zorder | hilbert}",
+      "temporalResolution": 86400000
+    },
+    "cellSize": {
+      "width": 0.5,
+      "height": 0.5
+    },
+    "partitions": 5000 // optional // tile into some layout scheme
+  }
+
+* *maxZoom* — max zoom level [optional field]
+* *tileSize* — destination tile size [optional field]
+* *resampleMethod* — ``{nearest-neighbor | bilinear | cubic-convolution | cubic-spline | lanczos}`` methods are possible
+* *keyIndexMethod:*
+    * *type* — ``{zorder | hilbert}``
+    * *temporalResolution* — temporal resolution in ms, if specified it would be a temporal index [optional field]
+* *cellSize* — cellSize [optional field]
+* *partitions* — partitions number after tiling
+
+*Pyramid definition:*
+
+.. code:: javascript
+
+  {
+    "type": "pyramid"
+  }
+
+*Render definition:*
+
+.. code:: javascript
+
+  {
+    "type": "render",
+    "format": "{tiff | png}",
+    "path": "{path | pattern}"
+  }
+
+* *format* — ``{tiff | png}`` supported formats
+* *path* — ``{path | pattern}`` output path, can be specified as a pattern
+
+*{Save | Update | Reindex} definition:*
+
+.. code:: javascript
+
+  {
+    "type": "{save | update | reindex}",
+    "name": "layer name",
+    "backend": {
+      "path": "path or table",
+      "profile": "profile name"
+    }
+  }
+
+* *name* — layer name, all inputs would be saved / updated / reindexed with that name
+* *backend:*
+    * *path* — path or table name
+    * *profile* — profile name, can be specified in the ``backend-profiles.json``, default profiles available: ``{file | hadoop | s3}``
+
+*Pipeline example:*
+
+.. code:: javascript
+
+  [
+    {
+      "type": "reproject",
+      "method": "{buffered | per-tile}",
+      "crs": "{EPSG code | EPSG name | proj4 string}",
+      "cellSize": {
+        "width": 0.5,
+        "height": 0.5
+      }
+    },
+    {
+      "type": "tile",
+      "maxZoom": 19,
+      "tileSize": 256,
+      "resampleMethod": "bilinear",
+      "layoutScheme": "zoomed",
+      "keyIndexMethod": {
+        "type": "zorder",
+        "temporalResolution": 86400000
+      },
+      "partitions": 5000
+    },
+    {
+      "type": "pyramid"
+    },
+    {
+      "type": "render",
+      "format": "{tiff | png}",
+      "path": "{path | pattern}"
+    },
+    {
+      "type": "{save | reindex | update}",
+      "name": "layer name",
+      "backend": {
+        "path": "path or table",
+        "profile": "profile name"
+      }
+    }
+  ]
+
+Conclusion
+^^^^^^^^^^
+
+Current ``input.json`` and ``backend-profiles.json`` are seems to be already fine. Significant changes should be introduced
```

Comment:
`Current` -> `The current`

`are seems to be already fine` -> `seem to be fine as-is`